src/flatfiledirmgr.rs

```diff
@@ -70,13 +80,36 @@ impl<R: Runtime> FlatFileDirMgr<R> {
         }))
     }
 
+    /// Check cache directory content.
+    fn check_directory(cache_path: &Path) -> Result<()> {
+        let mut any_missing = false;
+        for filename in [
+            CONSENSUS_FILENAME,
+            MICRODESCRIPTORS_FILENAME,
+            CERTIFICATE_FILENAME,
+            CHURN_FILENAME,
+        ]
+        .iter()
+        {
+            if !cache_path.join(filename).exists() {
+                any_missing = true;
+                debug!("required file missing: {filename}");
+            }
+        }
+        if any_missing {
+            return Err(Error::CacheCorruption("required files missing in cache"));
+        }
+        Ok(())
```

Comment:
```suggestion
        let missing_files = [
            CONSENSUS_FILENAME,
            MICRODESCRIPTORS_FILENAME,
            CERTIFICATE_FILENAME,
            CHURN_FILENAME,
        ]
            .iter()
            .filter(|filename| !cache_path.join(filename).exists())
            .collect();
        
        if !missing_files.is_empty() {
            debug!("required file(s) missing: {missing_files}");
            return Err(Error::CacheCorruption("required file(s) '{missing_files}' missing in cache"));
        }
        Ok(())
```

How about something like this? So you don't need the temporary variable `any_missing`, and can pass all the missing files to the caller.

If you think this is not nice, you should at least remove the `iter` on line 92, as it's not needed.

You can also go full-iterator and do it like this:

```
        [
            CONSENSUS_FILENAME,
            MICRODESCRIPTORS_FILENAME,
            CERTIFICATE_FILENAME,
            CHURN_FILENAME,
        ]
            .iter()
            .all(|filename|
                !cache_path.join(filename).exists().then(|| {
                    debug!("required file missing: {filename}");
                }).is_none()
            )
            .then(|| ())
            .ok_or_else(|| Error::CacheCorruption("required files missing in cache"))
```

But that would probably go into your "rust is unreadable" section :) And it took me 30 minutes to figure it out. I really got rusty.